rust/lance/src/index/vector/ivf/v2.rs

```diff
@@ -291,13 +291,22 @@ impl<S: IvfSubIndex + 'static, Q: Quantization + 'static> Index for IVFIndex<S,
 
         let centroid_vecs = centroids_to_vectors(self.ivf.centroids.as_ref().unwrap())?;
 
+        let index_type = match self.sub_index_type() {
+            (SubIndexType::Flat, QuantizationType::Flat) => "IVF_FLAT".to_owned(), // we don't want double FLAT
+            (sub_index_type, quantization_type) => {
+                format!("IVF_{}_{}", sub_index_type, quantization_type)
+            }
```

Comment:
What is this for IVF PQ? `IVF_FLAT_PQ`? Could we have a unit test for that?

Reply:
We are referring to this as IVF_PQ in many places in code. Do we want to keep it as IVF_PQ or introduce a new IVF_FLAT_PQ value?

Reply:
yes I noticed this, for now v2 doesn't support `IVF_PQ` so here it won't reach this.
`IVF_PQ` will be `IVF_PQ_PQ`, it's `IVF_PQ_PQ` because search PQ is not a trivial scan (what FLAT index does), so we need a `PQIndex`.
additionally, will simplify it to `IVF_PQ` like what we do here to `IVF_FLAT`